Common/Securities/IBuyingPowerModel.cs

```diff
@@ -37,6 +37,26 @@ public interface IBuyingPowerModel
         /// <param name="leverage">The new leverage</param>
         void SetLeverage(Security security, decimal leverage);
 
+        /// <summary>
+        /// Gets the margin currently allocated to the specified holding
+        /// </summary>
+        /// <param name="parameters">An object containing the security and holdings quantity/cost/value</param>
+        /// <returns>The maintenance margin required for the provided holdings quantity/cost/value</returns>
+        MaintenanceMargin GetMaintenanceMargin(MaintenanceMarginParameters parameters);
```

Comment:
TODO: 
- Make `MaintenanceMargin` & `InitialMargin` use a `List<CashAmount>` instead of a plane `decimal`
- `CashBuyingPM` implement `InitialMargin` methods so that it returns in the actual currency required, not in account currency
- Rename `Margin` for `BuyingPower`
- If `GetReservedBuyingPowerForPosition` supported a provided quantity instead of just using `Holdings`, we can remove `GetMaintenanceMargin` from interface
- Switch `GetInitialMarginRequirement` to protected again, if possible
- `HasSufficientBuyingPowerForOrder` could include the initial margin requirement for order in the response and we can get rid of `GetInitialMarginRequiredForOrder`